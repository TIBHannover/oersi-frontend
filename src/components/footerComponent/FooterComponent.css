.footer {
  background-color: transparent;
  margin-top: 6%;
}
.footer-block > h5 {
  color: #a9a9a9;
  font-size: 1.1rem;
}
.footer-block > hr {
  border: 1px solid rgba(0, 0, 0, 0.1);
  width: 50%;
  margin-left: unset;
}

.fotter-p {
  color: #a9a9a9;
  width: 250px;
}
.footer-block > ul {
  padding: 0px;
  list-style: none;
  display: block;
  float: left;
  line-height: 30px;
  width: 260px;
}

.img-block {
  height: 60px;
}
