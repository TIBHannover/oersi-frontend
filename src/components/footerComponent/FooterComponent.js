import React, {useState, useEffect} from "react"
import "./FooterComponent.css"
import {withTranslation} from "react-i18next"
import {getConfiguration} from "../../service/configuration/configurationService"
import PropTypes from "prop-types"
import parse from "html-react-parser"

/**
 * This is the Footer component, You can use different url and image after Build
 * use Fetsch to call public/footer/config.json to load data
 * @author Edmond Kacaj <edmondikacaj@gmail.com>
 */

const FooterComponent = () => {
  const [data, setdata] = useState("")
  const [isLoaded, setisLoaded] = useState(false)

  useEffect(() => {
    async function fetchData() {
      const res = await getConfiguration("/footer/footer.html")
      const htmlResponse = await res.text()
      if (
        !htmlResponse.includes("html") ||
        !htmlResponse.includes("head") ||
        !htmlResponse.includes("body")
      ) {
        setdata(htmlResponse)
        setisLoaded(true)
      } else {
        setisLoaded(false)
      }
    }
    fetchData()
  }, [])

  return <div data-insert-template-id="footer-id">{isLoaded && parse(data)}</div>
}

FooterComponent.propTypes = {
  data: PropTypes.object,
  isLoaded: PropTypes.bool,
}

export default withTranslation()(FooterComponent)
