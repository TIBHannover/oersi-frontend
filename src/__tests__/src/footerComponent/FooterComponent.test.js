import FooterComponent from "../../../components/footerComponent/FooterComponent"
import React from "react"
import {mount, shallow} from "../../../setupFiles"

describe("FooterComponent ==> UI  ", () => {
  let wrapperShadow
  beforeEach(() => {
    wrapperShadow = shallow(<FooterComponent />)
  })

  it("FooterComponent : should render correctly", () => {
    expect(wrapperShadow).toMatchSnapshot()
  })

  it(" FooterComponent:  wraps content in a div with .copyright pull-right", () => {
    expect(wrapperShadow.find("footer").length).toBe(1)
  })

  it(" FooterComponent:  Find Text Blog inside  Component", () => {
    expect(wrapperShadow.text().includes("Blog")).toBe(true)
  })
  it(" FooterComponent:  Find Text Datenschutz  inside Component", () => {
    expect(wrapperShadow.text().includes("Datenschutz")).toBe(true)
  })
})
