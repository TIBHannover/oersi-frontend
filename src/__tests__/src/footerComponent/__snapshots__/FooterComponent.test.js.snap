// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent ==> UI   FooterComponent : should render correctly 1`] = `ShallowWrapper {}`;
